src/span_context.js

```diff
@@ -128,26 +129,29 @@ export default class SpanContext {
             return null;
         }
 
-        let traceId = parseInt(headers[0], 16);
-        let NaNDetected = (isNaN(traceId, 16) || traceId === 0) ||
-                          isNaN(parseInt(headers[1], 16)) ||
-                          isNaN(parseInt(headers[2], 16)) ||
-                          isNaN(parseInt(headers[3], 16));
+        let traceId = Utils.isParsableHex64(headers[0]);
+        let spanId = Utils.isParsableHex64(headers[1]);
+        let parentId = Utils.isParsableHex64(headers[2]);
+        let flags = parseInt(headers[3], 16);
+        let NaNDetected = ((isNaN(traceId) || traceId.toNumber() === 0) ||
+                          isNaN(spanId)   ||
+                          isNaN(parentId) ||
+                          isNaN(flags));
 
         if (NaNDetected) {
             return null;
         }
 
-        let parentId = null;
-        if (headers[2] !== '0') {
-            parentId = Utils.encodeInt64(headers[2]);
+        // The collector expects null instead of zero.
+        if (parentId.toNumber() === 0) {
```

Comment:
what happens if parent ID is not 0 but a number > 2^53? Will toNumber() work still?  It seems like we can just check if the string was 0 to begin with.